apps/web/src/components/TweetInfo.tsx

```diff
@@ -65,13 +79,58 @@ const TweetInfo = ({ tweet }: TweetInfoProps) => {
                 {`@${tweet.userInfo.username}`}
               </Text>
             </Box>
-          </Flex>
-          <Text mt={2}>{decodeHTML(tweet.text)}</Text>
-          <Text color="gray.500" marginTop={2}>
-            {tweetTimeInfo()}
-          </Text>
+          </Link>
         </Flex>
-      </Link>
+        <Box zIndex={1} pointerEvents="none">
+          <Text mt={2}>
+            {parseContent(decodeHTML(tweet.text)).map((chunk) => {
+              if (chunk.type === 'text') return chunk.value;
+
+              if (chunk.type === 'url') {
```

Comment:
what about 1 component per chunk type?

<TweetChunkUrl />

Reply:
try to follow this pattern https://medium.com/reactbrasil/formik-como-criar-um-formul%C3%A1rio-com-campos-din%C3%A2micos-a150d60a8610